src/mouse_events.cpp

```diff
@@ -624,12 +624,12 @@ bool mouse_handler::mouse_button_event(const SDL_MouseButtonEvent& event, uint8_
 
 	if (gui().view_locked() || button < SDL_BUTTON_LEFT || button > buttons.size()) {
 		return false;
-	} else if (event.state > SDL_PRESSED || !gui().get_map().on_board(loc)) {
+	} else if (event.down || !gui().get_map().on_board(loc)) {
```

Comment:
Is this right? I don't think the first condition was ever valid, unless `state` could be something other than `SDL_Pressed` (which is 1) or `SDL_Released` (which is 0).

Reply:
updated.